packages/atlas/src/joystream-lib/extrinsics.ts

```diff
@@ -1163,4 +1163,47 @@ export class JoystreamLibExtrinsics {
 
     return { block }
   }
+
+  purchaseTokenOnMarketTx = async (tokenId: string, memberId: string, amount: string) => {
+    const amountCast = createType('u128', new BN(amount))
+
+    return this.api.tx.projectToken.buyOnAmm(
+      parseInt(tokenId),
+      parseInt(memberId),
+      amountCast,
+      createType('Option<ITuple<[Permill, u128]>>', [createType('Permill', new BN(0.5)), amountCast]) // percent, number of joy user wants to pay --- default on 0.5%
```

Comment:
same thing that @thesan mentioned in #4946

Reply:
answered in mentioned PR

Reply:
Well, it will be addressed for both transactions in #4946, so this can be considered resolved